ui/de-lib/src/main/java/org/iplantc/de/apps/client/presenter/categories/AppCategoriesPresenterImpl.java

```diff
@@ -93,192 +90,140 @@ public int compare(AppCategory group1, AppCategory group2) {
         return groupNames;
     }
 
     @Override
     public AppCategory getSelectedAppCategory() {
-        return view.getTree().getSelectionModel().getSelectedItem();
+        AppCategory selectedCategory = null;
+        Tree<AppCategory, String> tree = workspaceView.getTree();
+        if (tree != null) {
+            selectedCategory = tree.getSelectionModel().getSelectedItem();
+        }
+        return selectedCategory;
     }
 
     @Override
     public AppCategoriesView getView() {
-        return view;
+        return workspaceView;
     }
 
     @Override
-    public void go(final HasId selectedAppCategory) {
-        if (!treeStore.getAll().isEmpty()
-                && selectedAppCategory != null) {
-            AppCategory desiredCategory = treeStore.findModelWithKey(selectedAppCategory.getId());
-            view.getTree().getSelectionModel().select(desiredCategory, false);
-        } else {
-            view.mask(appearance.getAppCategoriesLoadingMask());
-            appService.getAppCategories(new AsyncCallback<List<AppCategory>>() {
-                @Override
-                public void onFailure(Throwable caught) {
-                    ErrorHandler.post(caught);
-                    view.unmask();
-                }
+    public void go(final HasId selectedAppCategory, final TabPanel tabPanel) {
+        this.viewTabPanel = tabPanel;
 
-                @Override
-                public void onSuccess(List<AppCategory> result) {
-                    addAppCategories(null, result);
-                    view.getTree().expandAll();
-                    if (selectedAppCategory != null) {
-                        AppCategory desiredCategory = treeStore.findModelWithKey(selectedAppCategory.getId());
-                        view.getTree().getSelectionModel().select(desiredCategory, false);
-                    } else {
-                        view.getTree().getSelectionModel().selectNext();
-                        view.getTree().getSelectionModel().select(treeStore.getRootItems().get(0), false);
-                    }
-                    view.unmask();
-                }
-            });
-        }
-    }
-
-    @Override
-    public void onAdd(StoreAddEvent<App> event) {
-        // When the list store adds
-        AppCategory appCategory = getSelectedAppCategory();
-        if (appCategory == null) {
-            return;
-        }
-        updateAppCategoryAppCount(appCategory, event.getSource().getAll().size());
-    }
+        viewTabPanel.add(workspaceView.getTree(), new TabItemConfig(appearance.workspaceTab()));
+        viewTabPanel.add(hpcView.getTree(), new TabItemConfig(appearance.hpcTab()));
 
-    @Override
-    public void onAppFavoriteSelected(AppFavoriteSelectedEvent event) {
-        final App app = event.getApp();
-        appUserService.favoriteApp(app, !app.isFavorite(), new AsyncCallback<Void>() {
+        workspaceView.getTree().mask(appearance.getAppCategoriesLoadingMask());
```

Comment:
Also create a method in the view called mask/unmask so we're not accessing the tree directly.